File: Data Structures & Algorithms/two-integer-sum/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
            int k=target-nums[i];
            if(mp.find(k)!=mp.end()&& mp[k] != i){
                return {i,mp[k]};
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-4.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res,sol=[],[]
        n=len(nums)

        def dfs(i,s):
            if s==target:
                res.append(sol.copy())
                return
            if s>target or i>=n:
                return
            sol.append(nums[i])
            dfs(i,s+nums[i])
            sol.pop()
            dfs(i+1,s)
            
        dfs(0,0)
        return res